client/lib/signup/flow-controller.js

```diff
@@ -4,36 +4,50 @@
  */
```

Comment:
This is a beast to review. Everything seemed okay on my first pass, but I'm still a bit wary of this module.